zetaclient/chains/evm/observer/observer.go

```diff
@@ -166,6 +167,11 @@ func FetchZetaTokenContract(
 
 // Start all observation routines for the evm chain
 func (ob *Observer) Start(ctx context.Context) {
+	if noop := ob.Observer.Start(); noop {
+		ob.Logger().Chain.Info().Msgf("observer is already started for chain %d", ob.Chain().ChainId)
```

Comment:
Will this log print a lot when the bootstrapper keep starting a chain that have already started? Just curious

Reply:
It calculated the diff only. New observers/signers won't start if present.
- https://github.com/zeta-chain/node/blob/2f06e5639910af4fbfeb4261ff0faf4a86b53af1/zetaclient/orchestrator/bootstrap.go#L152
- https://github.com/zeta-chain/node/blob/f2d14dfe0023cad7502ba4e2a663d2b81cd484a9/zetaclient/orchestrator/bootstap_test.go#L205

Reply:
Not sure, just curious. When will `ob.Observer.Start()` returns a `true` and print the log?

It seems the callback function `onAfterAdd` is called on a newly created observer on [this line](https://github.com/zeta-chain/node/blob/feat%2Fzetaclient-runtime-chains/zetaclient/orchestrator/bootstrap.go#L266), which contains a `false` on its `started` field.